studio/components/interfaces/Auth/AuthProvidersForm.tsx

```diff
@@ -0,0 +1,367 @@
+  IconChevronUp,
+  Input,
+  InputNumber,
+  Listbox,
+  Toggle,
+} from '@supabase/ui'
+import { FormHeader } from 'components/ui/Forms'
+import { HorizontalShimmerWithIcon } from 'components/ui/Shimmers'
+import { useStore } from 'hooks'
+import { observer } from 'mobx-react-lite'
+import { useState } from 'react'
+import ReactMarkdown from 'react-markdown'
+import { PROVIDERS_SCHEMAS } from 'stores/authConfig/schema'
+
+interface Enum {
+  label: string
+  value: string
+  icon: string
+}
+interface Provider {
+  $schema: string
+  type: 'object'
+  title: string
+  properties: {
+    [x: string]: {
+      title: string
+      type: 'boolean' | 'string' | 'select' | 'number'
+      description?: string
+      descriptionOptional?: string
+      enum: Enum[]
+      show: {
+        key: string
+        matches: string
+      }
+    }
+  }
+  validationSchema: any // todo: use Yup type
+  misc: {
+    iconKey: 'gitlab-icon'
+    requiresRedirect: true
+    helper: string
+    alert: {
+      title: string
+      description: string
+    }
+  }
+}
+
+const ProviderCollapsibleClasses = [
+  'bg-scale-100 dark:bg-scale-300 ',
+  'hover:bg-scale-200 dark:hover:bg-scale-500',
+  'data-open:bg-scale-200 dark:data-open:bg-scale-500 ',
+  'border-scale-300 ',
+  'dark:border-scale-500 hover:border-scale-500 ',
+  'dark:hover:border-scale-700 data-open:border-scale-700',
+  'data-open:pb-px col-span-12 mx-auto',
+  '-space-y-px overflow-hidden',
+  'border shadow',
+  'transition',
+  'first:rounded-tl',
+  'first:rounded-tr',
+  'last:rounded-bl',
+  'last:rounded-br',
+  'hover:z-50',
+]
+
+const AuthProvidersForm = () => {
+  const providers = PROVIDERS_SCHEMAS
+  const { authConfig } = useStore()
+
+  return (
+    <div>
+      <FormHeader
+        title="Auth Providers"
+        description={`URLs that auth providers are permitted to redirect to post authentication`}
+      />
+
+      <div className="-space-y-px">
+        {!authConfig.isLoaded
+          ? // true
+            providers.map((i) => (
+              <div className={[...ProviderCollapsibleClasses, 'px-6 py-3'].join(' ')}>
+                <HorizontalShimmerWithIcon />
+              </div>
+            ))
+          : // @ts-expect-error
+            // to do: fix type error, needs to be dynamic
+            providers.map((provider: Provider, i) => {
+              return <ProviderForm provider={provider} key={i} />
+            })}
+      </div>
+    </div>
+  )
+}
+
+const ProviderForm = ({ provider }: { provider: Provider }) => {
+  const [open, setOpen] = useState(false)
+  const { authConfig, ui } = useStore()
+  const active = authConfig.config[`EXTERNAL_${provider?.title?.toUpperCase()}_ENABLED`]
+  const INITIAL_VALUES: { [x: string]: string } = {}
+
+  /**
+   * construct values for INITIAL_VALUES
+   *
+   * return empty string `""` rather than `null`
+   * as it breaks form. null is not a valid value.
+   */
+  Object.keys(provider.properties).forEach((key) => {
```

Comment:
wrap this in a useEffect? otherwise it'll always be run on every re-render

Reply:
I think that might be the point, i'll check though